From b14a762e9940739df52c81e6bc888cc01579ed46 Mon Sep 17 00:00:00 2001
From: Edward E <develinthedetail@gmail.com>
Date: Fri, 1 Dec 2017 12:32:28 -0600
Subject: [PATCH] icontheme (win32): detect SVG files by extension if
 unregistered

It seems XP shipped without a registered Content Type for .svg

https://bugzilla.gnome.org/show_bug.cgi?id=781020
---
 gtk/gtkicontheme.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gtk/gtkicontheme.c b/gtk/gtkicontheme.c
index 532d94b..98cb5f2 100644
--- a/gtk/gtkicontheme.c
+++ b/gtk/gtkicontheme.c
@@ -2927,6 +2927,11 @@ icon_info_ensure_scale_and_pixbuf (GtkIconInfo  *icon_info,
 
               if (mime_type && strcmp (mime_type, "image/svg+xml") == 0)
                 is_svg = TRUE;
+#ifdef G_OS_WIN32
+              else
+                if (mime_type && strcmp (mime_type, "application/x-ext-svg") == 0)
+                  is_svg = TRUE;
+#endif
             }
 
           g_object_unref (file_info);
-- 
2.9.1

